### PR TITLE
Tweaks related to Camera2D drag margins

### LINKED ISSUE #23108: No way to set starting Camera2D offset within Drag Margin
**Godot version:**

3.1-alpha b550f93

**OS/device including version:**

Linux Mint 19 Cinnamon x64

**Issue description:**

When you have a Camera2D in your scene, it will always start dead-center on its parent node, regardless of whether Drag Margin H/V is enabled or not. It would be nice to be able to define the camera's starting offset from its parent node, relative to the drag margins. The only way to get this functionality now is via script:

```gdscript
func _ready():
	# disable drag margins
	$Camera2D.drag_margin_h_enabled = false
	$Camera2D.drag_margin_v_enabled = false
	
	# set drag margin offset
	$Camera2D.offset_h = 1
	$Camera2D.offset_v = 1
	
	# re-enable drag margins
	$Camera2D.drag_margin_h_enabled = true
	$Camera2D.drag_margin_v_enabled = true
```

Both `offset_h` and `offset_v` are already exposed to the inspector, but do nothing if their respective axes's drag margins are enabled (they DO shift the camera gizmos though...) It would be nice if they were to act as a starting offset in this case.

**Steps to reproduce:**
1. Add a Camera2D node as a child of a Node2D-derived node (such as Sprite)
2. Check both "Drag Margin H enabled" and "Drag Margin V Enabled" checkboxes
3. Check the "Draw Drag Margin" checkbox
4. Change both Offset values ("H" and "V") and watch as the camera rectangles shift properly (never letting the parent node outside of the drag margin rectangle)
5. Play the scene; the camera will go back to being centered on its parent node

### LINKED ISSUE #26102: Keep Camera2D's Drag MarginH/V Enabled off by default
**Godot version:**

aa5b99821b23d74eafb49f4b0d2d86fe693a903c

**Issue description:**

On more than one occasion, while assisting people in the discord server, someone will come in wondering why, no matter what they do, their camera2D simply will not move.  After going through a number of troubleshooting questions, with changing just about everything possible, with recreating the scripts and nodes and setting up signals, and questioning whether Godot is just experiencing a highly evasive bug, somehow the camera just will not move.

Then, seemingly miraculously, it works, by just letting the script run for a while; then it becomes clear: oh right, the drag margins.

In my view, these properties being on by default does more harm than good; I don't believe the typical user, when setting up their camera2D, wants these properties to be on, in fact I think it's a niche need for particular kinds of games, where other games usually just want the camera to follow as the developer dictates it.  Given the problems it causes with people coming into the engine and becoming near-immediately frustrated, and the fact that you probably don't need the drag margins on until you understand what they do, I think they should default to off; I believe this will spare a lot of headaches, and doesn't take but two clicks for those who need them on.

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes #26102
Fixes #23108

Most important change is that settings H/V offset now forces the camera position to update if drag margins are enabled. It can be used to set it's initial offset and I added some lines to docs that clarify the usage.

From other changes, drag margins are disabled by default as requested in first issue.
I also swapped places of V Offset and H Offset, as usually horizontal stuff comes before vertical one in inspector and that inconsistency was triggering me >.>

### CODE REVIEW

**Comment by samgreen** [comment]
Curious. Since we're no longer serializing default values like drag margin, does this mean all existing users will have to manually toggle the status of their `h_drag_enabled` and `v_drag_enabled` in their inspector once this patch is applied?

**Comment by KoBeWi** [comment]
Yeah, that's a slight compatibility breakage.
Users who didn't use drag margins won't be affected, aside from two properties getting removed in .tscn files with Camera2D.
Users who used the drag margins will now need to enable them manually. But that's not much work and if this comes into changelog, affected users will know to adapt their projects on upgrade to 3.2.

tl;dr not really a big deal

File: doc/classes/Camera2D.xml [hunk 1]
```diff
@@ -120,9 +120,11 @@
 		</member>
 		<member name="offset_h" type="float" setter="set_h_offset" getter="get_h_offset">
 			The horizontal offset of the camera, relative to the drag margins. Default value: [code]0[/code]
+			[b]Note:[/b] Offset H is used only to force offset relative to margins. It's not updated in any way if drag margins are enabled and can be used to set inital offset.
 		</member>
 		<member name="offset_v" type="float" setter="set_v_offset" getter="get_v_offset">
 			The vertical offset of the camera, relative to the drag margins. Default value: [code]0[/code]
+			[b]Note:[/b] Used the same as Offset H.
```
[the remark this answers is not in the document]
Comment (KoBeWi) [reply doc/classes/Camera2D.xml]: Forgot about that. Fixed.

**Review by akien-mga — APPROVED** [review]
Should be ok to merge once rebased.
The compat change needs to be clearly mentioned in the changelog @Calinou.

**Comment by KoBeWi** [comment]
Rebased.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Calinou** [comment]
@akien-mga Added to the changelog :slightly_smiling_face:
